src/main/java/dev/blaauwendraad/masker/json/ValueMaskers.java

```diff
@@ -394,7 +394,7 @@ public static ValueMasker.AnyValueMasker withTextFunction(Function<String, @Null
                                                 // surrogates which together form one unicode character.
                                                 int codePoint = -1;
                                                 if (Character.isHighSurrogate(unicodeHexBytesAsChar) // first surrogate must be the high surrogate
-                                                        && encodedIndex < context.byteLength() - 6 /* -6 for all bytes of
+                                                        && encodedIndex <= context.byteLength() - 6 /* -6 for all bytes of
```

Comment:
Found this when copying the code into `KeyMatcher` and the latest encoded character wasn't matching properly due to incorrect condition. In this particular place it doesn't change anything because the latest character is always `"` so that condition is always true for the valid strings

Reply:
Perhaps it's one more reason to adhere to jackson behavior in `withTextFunction` as well